src/components/Practice/Mechanics7.jsx

```
@@ -0,0 +1,877 @@
+  audio,
+  currentImg,
+}) => {
+  const [words, setWords] = useState(
+    type === "word" ? [] : ["Friend", "She is", "My"]
+  );
+  const [recordingStates, setRecordingStates] = useState({});
+
+  useEffect(() => {
+    if (words && words.length) {
+      setRecordingStates(
+        words.reduce((acc, word) => ({ ...acc, [word]: false }), {})
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)